packages/block-editor/src/components/block-popover/index.js

```diff
@@ -74,6 +77,9 @@ function BlockPopover(
 		};
 	}, [ selectedElement ] );
 
+	const { isZoomOut: isZoomOutFn } = unlock( useSelect( blockEditorStore ) );
+	const isZoomOut = isZoomOutFn();
```

Comment:
Always call selector functions inside the callback.

This will ensure there is a subscription to the store so that when the state of the zoom out changes it will re-render the component.


```suggestion
	
const { isZoomOut } = useSelect(
  (select) => ({
    isZoomOut: unlock(select(blockEditorStore)).isZoomOut(),
  }),
  []
);

```

Reply:
Do we need a subscription? Is anything that doesn't work if we don't have one?

Reply:
Possibly not right now. But code can change.

My understanding is as follows but I'd be happy to be advised I'm wrong.

If you don't subscribe to _changes_ to the zoom out state then this component may provide a _stale_ version of `const isZoomOut`. Thus when your `if( isZoomOut )` condition is run it _might_ run with the wrong (stale) value.

Currently you are relying on a set of unrelated conditions to hope that the calculation for the popover is re-run if/when the zoom out changes, rather than actually coupling it to the state change of zoom out.

More information see `Always call selector functions inside the callback` from:

https://jardasn.blog/2021/01/22/some-best-practices-for-using-useselect-from-wordpress-data/

Reply:
I was also aware (possibly mistakenly) of some consideration around subscriptions impacting performance, and when this mode is toggled on and off via UI affordance maybe we can get away w/o one. OTOH one can set this mode programatically, while the user took no action 🤷🏻

Reply:
You are right to be concerned regarding subscriptions and performance. That said, this particular `useSelect` is only subscribing to a single portion of state. That state will only change very infrequently - users don't often _rapidly_ toggle between editor zoom levels.

As a result my instinct is that it's more important that the popover position is calculated correctly when the zoom changes so it's ok to have the subscription.

Reply:
cc @jsnajdr who wrote the article linked above

Reply:
It seems that the `isZoomOut` value is only ever true if we're inside the `ZoomOutPopover` component. Then we want to align the popover differently, taking into account the parent element. Then the value is really constant during the entire lifetime of the component, because `ZoomOutPopover` itself is rendered conditionally only when `isZoomOut` is true.

So, things still work the same if we don't do a subscription. But I would still recommend to do it, because we're always one refactor away from breaking the original assumptions, and suddenly we end up with buggy code where the block toolbar is sometimes not aligned properly.

Subscriptions are bad for performance only when there are many of them. Like if we had an `isZoomOut` subscription for every block on the page, selected or not. But that's not true for `ZoomOutPopover`, is it? It's always rendered only once, on the selected block.

Reply:
Good to know my reasoning was sound 👍 Thanks @jsnajdr.

I agree entirely on the "we're only one refactoring away" argument.